### PR TITLE
Sparkswap/fetch orderbooks

### PR DESCRIPTION
**Pull request by dannypaz**
This PR adds `fetchOrderbook` call to the sparkswap ccxt exchange.

### CODE REVIEW

**Comment by dannypaz** [comment]
[add fetchOrderBook (non-streaming) to ccxt implementation](https://trello.com/c/0vOPudkI/755-add-fetchorderbook-non-streaming-to-ccxt-implementation)

File: js/sparkswap.js [hunk 1]
```diff
@@ -233,8 +232,24 @@ module.exports = class sparkswap extends Exchange {
         return this.privateGetV1OrderId ({ id });
     }
 
-    async fetchOrderBook () {
-        throw new ExchangeError ('Not Implemented');
+    async fetchOrderBook (market, params = {}) {
+        await this.loadMarkets ();
```
Comment (martineehrlich) [inline js/sparkswap.js:278]: symbol

File: js/sparkswap.js [hunk 2]
```diff
@@ -233,8 +232,24 @@ module.exports = class sparkswap extends Exchange {
...
+        //
+        const res = await this.privateGetV1Orderbook ({ 'market': this.marketId (market) });
+        // We remove the nanoseconds from the millisecond timestamp to fit the CCXT
+        // format
+        const millisecondTimestamp = res.timestamp.substring (0, res.timestamp.length - 6);
+        // We remove the nanoseconds (4 characters) from the ISO nanosecond timestamp to fit
+        // the ccxt format
+        const millisecondDatetime = res.datetime.substring (0, res.datetime.length - 5) + 'Z';
+        return {
+            'asks': res.asks,
```
Comment (martineehrlich) [inline js/sparkswap.js:290]: Have you checked that bids and asks are formatted correctly? I think they are but just checking

**Review by martineehrlich — CHANGES_REQUESTED** [review]

File: js/sparkswap.js (hunk 2 above)
Comment (dannypaz) [reply js/sparkswap.js:290]: Yup. we are all good.
screen shot 2018-10-03 at 4 06 42 pm

**Review by martineehrlich — APPROVED** [review]
LGTM!

**Pull request merged**